app/services/page_cloner.rb

```diff
@@ -0,0 +1,79 @@
+class PageCloner
+  def self.clone!(page)
+    new(page).clone!
+  end
+
+  def initialize(page)
+    @page = page
+  end
+
+  def clone!
+    cloned_page = @page.dup
+
+    # TODO: I'm thinking to have the user of this API
+    # decide what the new title should be and have it
+    # passed in to clone! It's not this class' responsibility
+    # to do the necessary checks against AK
```

Comment:
I think it makes sense to pass in the title, but like Tuuli I am worried that campaigners will want to have the same display title. I also am wondering where the AK check should happen, since it does need to.